scanpipe/pipes/scancode.py

```diff
@@ -310,18 +311,108 @@ def scan_for_files(project):
 
 def scan_for_application_packages(project):
     """
-    Runs a package scan on files without a status for a `project`.
+    Runs a package scan on files without a status for a `project`,
+    then create DiscoveredPackage and DiscoveredDependency instances
+    from the detected package data
 
     Multiprocessing is enabled by default on this pipe, the number of processes can be
     controlled through the SCANCODEIO_PROCESSES setting.
     """
     resource_qs = project.codebaseresources.no_status()
+
+    # Collect detected Package data and save it to the CodebaseResource it was
+    # detected from
     _scan_and_save(
         resource_qs=resource_qs,
         scan_func=scan_for_package_data,
         save_func=save_scan_package_results,
     )
 
+    # Iterate through CodebaseResources with Package data and handle them using
+    # the proper Package handler from packagedcode
+    assemble_packages(project=project)
```

Comment:
How are the performances of the new assemble? Do we have some stats about the time spent to assemble compared to the scan_and_save processing?

Reply:
I don't have any metrics on the performance impact that `assemble_packages` has on application package scanning. What is the process you do to collect performance metrics on different scancode.io parts?

Reply:
We run the pipeline using various inputs pre-changes then again post-changes and we can compare the execution time of the relevant steps.

Also, a profiling utility is available to go deeper into optimizations:  https://github.com/nexB/scancode.io/blob/main/scanpipe/pipelines/__init__.py#L155

Reply:
> How are the performances of the new assemble? Do we have some stats about the time spent to assemble compared to the scan_and_save processing?

The new assemble is a must have functional change that we should get working and tested before we can optimize if we ever need to optimize? ... @tdruez did you experience an issue?